modules/text-input/react/spec/TextInput.spec.tsx

```diff
@@ -1,50 +1,72 @@
-  test('Text Input in a FormField should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <FormField
-        label="My Field"
-        inputId="my-input-field"
-        hintText="Helpful text to resolve error"
-        hintId="my-input-field-error"
-        error={FormField.ErrorType.Error}
-      >
-        <TextInput placeholder="Placeholder" value={'Hello'} onChange={jest.fn()} />;
-      </FormField>
-    );
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when rendered', () => {
+    describe('with an placeholder', () => {
+      it('should render a text input with placeholder', () => {
+        const {getByPlaceholderText} = render(
+          <TextInput onChange={cb} placeholder={placeholder} />
+        );
+        expect(getByPlaceholderText(placeholder)).toHaveAttribute('placeholder', placeholder);
+      });
+    });
+
+    describe('with a id', () => {
+      it('should render a text input with a id', () => {
+        const {getByPlaceholderText} = render(
+          <TextInput id={id} onChange={cb} value={value} placeholder={placeholder} />
+        );
+        expect(getByPlaceholderText(placeholder)).toHaveAttribute('id', id);
+      });
+    });
+
+    describe('with a value', () => {
+      it('should render a text input with a value', () => {
+        const {getByDisplayValue} = render(<TextInput onChange={cb} value={value} />);
+        expect(getByDisplayValue(value)).toBeDefined();
+      });
+    });
+
+    describe('with disabled attribute', () => {
+      it('should render a disabled text input', () => {
+        const {getByDisplayValue} = render(
+          <TextInput onChange={cb} disabled={true} value={value} />
+        );
+        expect(getByDisplayValue(value)).toBeDisabled();
+      });
+    });
+
+    describe('with extra, arbitrary props', () => {
+      it('should spread extra props onto the text input', () => {
+        const attr = 'test';
+        const {getByPlaceholderText} = render(
+          <TextInput onChange={cb} data-propspread={attr} placeholder={placeholder} />
+        );
+        expect(getByPlaceholderText(placeholder)).toHaveAttribute('data-propspread', attr);
+      });
+    });
   });
 
-  test('Text Input with `aria-labelledby` should pass axe DOM accessibility guidelines', async () => {
-    const html = ReactDOMServer.renderToString(
-      <>
-        <label id="123">Label</label>
-        <TextInput
-          placeholder="Placeholder"
-          value={'Hello'}
-          aria-labelledby="123"
-          onChange={jest.fn()}
-        />
-        ;
-      </>
-    );
-    expect(await axe(html)).toHaveNoViolations();
+  describe('when provided an input ref', () => {
+    it('should render a text input with ref defined', () => {
+      const ref: React.RefObject<HTMLInputElement> = React.createRef();
+      render(<TextInput inputRef={ref} />);
+      expect(ref.current).toBeDefined();
```

Comment:
See this comment https://github.com/Workday/canvas-kit/pull/372#discussion_r366109323

According to React, `ref.current` will always be defined, we need to check if it's `null` or not. And if it's not, we should be checking if it is an input element.